blueocean-core-js/src/js/fetch.js

```diff
@@ -170,16 +194,28 @@ export const Fetch = {
      * @param {Object} [options.fetchOptions] - Optional isomorphic-fetch options.
      * @returns JSON body.
      */
-    fetchJSON(url, { onSuccess, onError, fetchOptions } = {}) {
+    fetchJSON(url, { onSuccess, onError, fetchOptions, disableCapabilites } = {}) {
+        let fixedUrl = url;
+        if (urlconfig.getJenkinsRootURL() !== '' && !url.startsWith(urlconfig.getJenkinsRootURL())) {
+            fixedUrl = `${urlconfig.getJenkinsRootURL()}${url}`;
+        }
+        let future;
         if (!config.isJWTEnabled()) {
-            return FetchFunctions.rawFetchJSON(url, { onSuccess, onError, fetchOptions });
+            future = FetchFunctions.rawFetchJSON(fixedUrl, { onSuccess, onError, fetchOptions });
+        } else {
+            future = jwt.getToken()
+                .then(token => FetchFunctions.rawFetchJSON(fixedUrl, {
+                    onSuccess,
+                    onError,
+                    fetchOptions: FetchFunctions.jwtFetchOption(token, fetchOptions),
+                }));
         }
-        return jwt.getToken()
-            .then(token => FetchFunctions.rawFetchJSON(url, {
-                onSuccess,
-                onError,
-                fetchOptions: FetchFunctions.jwtFetchOption(token, fetchOptions),
-            }));
+
+        if (!disableCapabilites) {
+            return future.then(data => capabilityAugmenter.augmentCapabilities(utils.clone(data)));
```

Comment:
The main reason I called clone was so that multple  augmenters were not updating the same object. I'm not sure thats an issue though.